File: reprex.R
library(wrapr)

from_point <- c(-77.0364, 38.8951)
to_point <- c(-78.0364, 39.8951)

dist <- 0.1
width <- 0.1
method <- "bilinear"
seg_id <- 1

#Must be in float64
ds <- "NE1_LR_LC.tif"

proj_str <- paste0(
  "+proj=tpeqd ",
  " +lon_1=", from_point[1], " +lat_1=",  from_point[2],
  " +lon_2=", to_point[1], " +lat_2=", to_point[2]
)
points_tpeqd <-
  sf::st_sfc(
    sf::st_point(from_point),
    sf::st_point(to_point),
    crs = "+proj=latlon"
  ) %.>%
  sf::st_transform(., proj_str)

point_1 <- points_tpeqd[[1]]
point_2 <- points_tpeqd[[2]]

bbox <- c(point_1[1], -(width * 0.5), point_2[1], (width * 0.5))

num_samples <- max(2, as.integer(ceiling((point_2[1] - point_1[1]) / dist)))

profile <- sf::gdal_utils(
  util = "warp",
  ds,
  "test_R.tif",
  config_options = c(
    dstSRS = proj_str,
    outputBounds = as.character(bbox),
    height = as.character(1),
    width = as.character(num_samples),
    resampleAlg = method
  )
)

File: reprex.py
from pyproj import CRS, Transformer
from osgeo import gdal
import pyproj.exceptions
import math
import multiprocessing
import alive_progress
import pathos.pools
import pandas
import shapely
import geopandas

gdal.UseExceptions()

from_point = [-77.0364, 38.8951]
to_point = [-78.0364, 39.8951]
ds = gdal.Open("NE1_LR_LC.tif")
dist = 0.1
width = 0.1
method = "bilinear"
seg_id = 1

proj_str = "+proj=tpeqd +lon_1={} +lat_1={} +lon_2={} +lat_2={}".format(
    from_point[0], from_point[1],
    to_point[0], to_point[1])

points_tpeqd = geopandas.GeoDataFrame(
  geometry = [
    shapely.Point(from_point),
    shapely.Point(to_point)
  ],
  crs = "+proj=latlon"
).to_crs(proj_str)

point_1 = points_tpeqd["geometry"][0].coords[0]
point_2 = points_tpeqd["geometry"][1].coords[0]

bbox = [point_1[0], -(width * 0.5), point_2[0], (width * 0.5)]

num_samples = max(2, int(math.ceil((point_2[0] - point_1[0]) / dist)))

profile = gdal.Warp(
  "test_py.tif",
  ds,
  dstSRS=proj_str,
  outputBounds=bbox,
  height=1,
  width=num_samples,
  resampleAlg=method
)